engine/crates/engine-v2/schema/src/builder/graph.rs

```diff
@@ -364,6 +384,46 @@ impl<'a> GraphBuilder<'a> {
             }
         }
 
+        // Adding all fully implemented interfaces per subgraph.
+        for interface_id in (0..self.graph.interface_definitions.len()).map(InterfaceDefinitionId::from) {
+            // Collecting per subgraph all objects defining this interface.
+            let mut found_ids: HashMap<SubgraphId, Vec<ObjectDefinitionId>> = HashMap::new();
```

Comment:
I don't think this algorithm isn't accurate. The following wouldn't be correctly taken into account:
```graphql
type Horse @join_type(graph: A) @join_implements(graph: A, interface: "Animal") implements Animal

type Dog @join_type(graph: B) @join_implements(graph: B, interface: "Animal") implements Animal

interface Animal
```

`Dog` is only available in subgraph `B`, so within `B` all the objects implement the interface `Animal` and there is no need to keep track of it.

IMHO the algorithm should do:
```
for interface in ... {
  let not_fully_implement_in_subgraphs: HashSet<SubgraphId> = HashSet::new();
  for object in interface.possible_types {
    for subgraph in object.subgraph {
        if does_not_implement_interface_in_subgraph(object, interface, subgraph) {
          not_fully_implement_in_subgraphs.insert(subgraph)
        }
    }
  }
}
```

Your algorithm is stricter, it requires that `Horse` implements `Animal` even in `B` where it's not present.